src/basic/fd-util.c

```diff
@@ -731,8 +731,7 @@ int fd_duplicate_data_fd(int fd) {
                 if (f != 0)
                         return -errno;
 
-                safe_close(copy_fd);
-                copy_fd = TAKE_FD(tmp_fd);
+                CLOSE_AND_REPLACE(copy_fd, tmp_fd);
```

Comment:
not sure if it shouldn't be lowercase...

i mean, there are no clear rules when to do things uppercase and when not. If something feels more macroish, then uppercase even if its actually an inline function, and if somethings feels more funcitonish then lowercase even if its actually a macro, i guess. But this case, what is it?

Reply:
I think if a macro does double-evaluation, then it's macro-ish. And TAKE_FD, TAKE_PTR, and this one do that...